src/riot-rs-embassy-common/src/identity.rs

```diff
@@ -0,0 +1,99 @@
+//! Tools and traits for describing device identities.
+//!
+//! See `riot_rs::identity` for general documentation.
+#![deny(missing_docs)]
+
+/// Trait describing the unique identifier available on a board.
+///
+/// See the module level documentation on the characteristics of the identifier.
+///
+/// # Evolution
+///
+/// In its current state, this type is mainly a wrapper around a binary identifier with a
+/// length constant at build time.
+///
+/// As it is used more, additional methods can be provided for concrete types of identifiers, such
+/// as MAC addresses. By default, those would be generated in some way from what is available in
+/// the identifier -- but boards where the identifier already *is* a MAC address (or possibly a
+/// range thereof) can provide their official addresses.
+pub trait DeviceId: Sized {
+    /// Some `[u8; N]` type, returned by [`.bytes()`][Self::bytes].
+    ///
+    /// This may not represent all the identifying information available on the board, but can
+    /// represent a unique portion thereof.
+    ///
+    /// (For example, if a device has two consecutive MAC addresses assigned, the type as a whole
+    /// may represent both, but the conventional serialized identity of the board may just be one
+    /// of them).
+    ///
+    /// # Evolution
+    ///
+    /// In the long run, it will be preferable to add a `const BYTES_LEN: usize;` and enforce the
+    /// type `[u8; Self::BYTES_LEN]` as the return value of [`.bytes(_)]`][Self::bytes]. This can
+    /// not be done yet as it depends on the `generic_const_exprs` featureVg
```

Comment:
```suggestion
    /// not be done yet as it depends on the `generic_const_exprs` feature.
```

Reply:
Can I take that as an indication that the direction is good enough to do a final pass of the docs before a final review and squash?